src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -570,15 +571,25 @@ void HMM<Distribution>::Forward(const arma::mat& dataSeq,
 
   arma::mat logTrans = trans(log(transition));
 
+  // Define a variable to store the value of log-probability for dataSeq.
+  arma::mat logProbs(dataSeq.n_cols, transition.n_rows);
+
+  // Save the values of log-probability to logProbs.
+  for (size_t i = 0; i < transition.n_rows; i++)
+  {
+    // Define arma::vec to store log-probability for any dataSeq column.
+    arma::vec logProb;
```

Comment:
I think we can do without using a temporary variable here.

Reply:
Right, agreed, I mentioned how in another comment above. :+1: